src/services/infra/DynamoDBService.ts

```diff
@@ -1,82 +1,45 @@
+import DynamoDBClient from "./DynamoDBClient"
 
-const MOCK_LAUNCH: SiteLaunchMessage = {
-  repoName: "my-repo",
-  appId: "my-app",
-  primaryDomainSource: "example.com",
-  primaryDomainTarget: "myapp.example.com",
-  domainValidationSource: "example.com",
-  domainValidationTarget: "myapp.example.com",
-  requestorEmail: "john@example.com",
-  agencyEmail: "agency@example.com",
-  githubRedirectionUrl: "https://github.com/my-repo",
-  redirectionDomain: [
-    {
-      source: "example.com",
-      target: "myapp.example.com",
-      type: "A",
-    },
-  ],
-  status: { state: "pending", message: "PENDING_DURING_SITE_LAUNCH" },
-}
 export default class DynamoDBService {
   private readonly dynamoDBClient: DynamoDBClient
 
   private readonly TABLE_NAME: string
 
-  constructor() {
-    this.dynamoDBClient = new DynamoDBClient()
+  constructor(dynamoDBClient: DynamoDBClient) {
+    this.dynamoDBClient = dynamoDBClient
     this.TABLE_NAME = config.get("aws.dynamodb.siteLaunchTableName")
     autoBind(this)
   }
 
   async createItem(message: SiteLaunchMessage): Promise<void> {
-    await this.dynamoDBClient.createItem(this.TABLE_NAME, MOCK_LAUNCH)
+    await this.dynamoDBClient.createItem(this.TABLE_NAME, message)
   }
 
-  async getItem(message: SiteLaunchMessage): Promise<GetCommandOutput> {
-    return this.dynamoDBClient.getItem(this.TABLE_NAME, MOCK_LAUNCH.appId)
-  }
+  async getAllSuccessOrFailureLaunches(): Promise<SiteLaunchMessage[]> {
+    const entries = ((await this.dynamoDBClient.getAllItems(this.TABLE_NAME))
+      .Items as unknown) as SiteLaunchMessage[]
+
+    const successEntries =
+      entries?.filter(
+        (entry) =>
+          entry.status?.state === "success" || entry.status?.state === "failure"
+      ) || []
 
-  async updateItem(message: SiteLaunchMessage): Promise<UpdateCommandOutput> {
-    // TODO: delete mocking after integration in IS-186
-    MOCK_LAUNCH.status = { state: "success", message: "SUCCESS_SITE_LIVE" }
-    const updateParams: UpdateParams = {
-      TableName: this.TABLE_NAME,
-      Key: { appId: MOCK_LAUNCH.appId },
-      // The update expression to apply to the item,
-      // in this case setting the "status" attribute to a value
-      UpdateExpression:
-        "set #status.#state = :state, #status.#message = :message",
-      ExpressionAttributeNames: {
-        "#status": "status",
-        "#state": "state",
-        "#message": "message",
-      },
-      // A map of expression attribute values used in the update expression,
-      // in this case mapping ":state" to the value of the Launch status state and ":message" to the value of the Launch status message
-      ExpressionAttributeValues: {
-        ":state": MOCK_LAUNCH.status.state,
-        ":message": MOCK_LAUNCH.status.message,
-      },
-    }
-    return this.dynamoDBClient.updateItem(updateParams)
+    // Delete after retrieving the items
+    Promise.all(successEntries.map((entry) => this.deleteItem(entry)))
```

Comment:
this might be dangerous - because we send the email out later on, failure in email sending leads to data loss and we won't know until the agency writes in

Reply:
Hmm, I dont think it's the responsibility of the dynamo db to store failed launches after they have been consumed by our backend. 

As a first cut we are also envisioning them to write in when there is a failure, what do you feel about treating this as an "unrecoverable" error for now, and in the future we can envision recovery options such as retires if the emails are unable to be sent?